src/nvim/globals.h

```diff
@@ -1039,6 +1039,10 @@ EXTERN int fill_vert INIT(= ' ');
 EXTERN int fill_fold INIT(= '-');
 EXTERN int fill_diff INIT(= '-');
 
+// Set when doing something for completion that may call edit() recursively,
+// which is not allowed. Also used to disable folding during completion
+EXTERN int compl_busy INIT(= false);
```

Comment:
can you move this under [compl_interrupted](https://github.com/nhooyr/neovim/blob/master/src/nvim/globals.h#L215), seems like that is where the compl-related variables live.